Tools/clinic/clinic.py

```diff
@@ -1988,13 +1988,30 @@ def parse_file(
     libclinic.write_file(output, cooked)
 
 
+def create_python_parser_namespace() -> dict[str, Any]:
+    ns = dict(
+        CConverter=CConverter,
+        CReturnConverter=CReturnConverter,
+        buffer=buffer,
+        robuffer=robuffer,
+        rwbuffer=rwbuffer,
+        unspecified=unspecified,
+    )
+    for name, converter in converters.items():
+        ns[f'{name}_converter'] = converter
+    for name, return_converter in return_converters.items():
+        ns[f'{name}_return_converter'] = return_converter
+    return ns
+
+
 class PythonParser:
     def __init__(self, clinic: Clinic) -> None:
-        pass
+        self.namespace = create_python_parser_namespace()
 
     def parse(self, block: Block) -> None:
+        ns = dict(self.namespace)
```

Comment:
It is already a dict, no?

Reply:
It's to create a copy, since each exec() call can modify the newly created `ns` dict.